glue/core/coordinate_helpers.py

```diff
@@ -53,9 +53,22 @@ def pixel2world_single_axis(wcs, *pixel, world_axis=None):
             pixel_new.append(p.flat[0])
     pixel = np.broadcast_arrays(*pixel_new)
 
-    result = wcs.pixel_to_world_values(*pixel)
+    # In the case of 1D WCS, there is an astropy issue which prevents us from
+    # passing arbitrary shapes - see https://github.com/astropy/astropy/issues/12154
+    # Therefore, we ravel the values and reshape afterwards
 
-    return broadcast_to(result[world_axis], original_shape)
+    if len(pixel) == 1 and pixel[0].ndim > 1:
+        pixel_shape = pixel[0].shape
+        pixel = [pixel[0].ravel()]
+        result = wcs.pixel_to_world_values(*pixel)
+        result = result.reshape(pixel_shape)
+    else:
+        result = wcs.pixel_to_world_values(*pixel)
+
+    if len(pixel) > 1:
```

Comment:
```suggestion
    if len(pixel) > 1 or np.shape(result)[0] == 1:
```
This fixes all tests for me. Of course this implies `world_axis == 0`, not sure if that should be explicitly tested as well or will always be satisfied...

Reply:
I think this is not the correct fix - I will push another commit up shortly (short version: some of our coordinate classes were not following APE 14 strictly)